### ISSUE TITLE
The num_pending or jetstream_consumer_num_pending value queried on the follow node of the consumer is incorrect.

### ISSUE THREAD

**Issue by yuzhou-nj** [nats-io/nats-server#4539]
### What version were you using?

nats-server: v2.9.22

### What environment was the server running in?

CentOS, Intel x86_64

### Is this defect reproducible?

Create a stream:
```
# nats s add test
? Subjects test.>
? Storage file
? Replication 3
? Retention Policy Interest
? Discard Policy Old
? Stream Messages Limit 1000
? Per Subject Messages Limit -1
? Total Stream Size -1
? Message TTL -1
? Max Message Size -1
? Duplicate tracking time window 2m0s
? Allow message Roll-ups No
? Allow message deletion Yes
? Allow purging subjects or the entire stream Yes
Stream test was created

```

Add a consumer:
```
# nats consumer add test
? Consumer name test_c1
? Delivery target (empty for Pull Consumers)
? Start policy (all, new, last, subject, 1h, msg sequence) all
? Acknowledgement policy explicit
? Replay policy instant
? Filter Stream by subject (blank for all)
? Maximum Allowed Deliveries -1
? Maximum Acknowledgements Pending 0
? Deliver headers only without bodies No
? Add a Retry Backoff Policy No
```

Pub a message. 
`nats pub "test.123" "hello"`

`nats consumer report test  ` ,  Unprocessed is 1:
```

╭────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                         Consumer report for test with 1 consumers                                          │
├──────────┬──────┬────────────┬──────────┬─────────────┬─────────────┬─────────────┬───────────┬────────────────────────────┤
│ Consumer │ Mode │ Ack Policy │ Ack Wait │ Ack Pending │ Redelivered │ Unprocessed │ Ack Floor │ Cluster                    │
├──────────┼──────┼────────────┼──────────┼─────────────┼─────────────┼─────────────┼───────────┼────────────────────────────┤
│ test_c1  │ Pull │ Explicit   │ 30.00s   │ 0           │ 0           │ 1 / 100%    │ 0         │ nats-55*, nats-64, nats-65 │
╰──────────┴──────┴────────────┴──────────┴─────────────┴─────────────┴─────────────┴───────────┴────────────────────────────╯
```
on the consumer leader node(here is nats-55),  num_pending is correct:

```
# curl "http://127.0.0.1:8222/jsz?consumers=1"
...
"num_pending": 1,
...
```

```
# curl -s "http://127.0.0.1:7777/metrics"  |grep test_c1 |grep jetstream_consumer_num_pending
jetstream_consumer_num_pending{account="USERS",cluster="nats_c",consumer_desc="",consumer_leader="nats-55",consumer_name="test_c1",domain="",is_consumer_leader="true",is_meta_leader="true",is_stream_leader="false",meta_leader="nats-55",server_id="http://x.x.x.55:8222",server_name="nats-55",stream_leader="nats-65",stream_name="test"} 1
```

But on nats-64 and nats-65, num_pending is zero.

```
# curl "http://127.0.0.1:8222/jsz?consumers=1"
...
          "consumer_detail": [
            {
              "stream_name": "test",
              "name": "test_c1",
...
              "num_waiting": 0,
              "num_pending": 0,
              "cluster": {
...


# curl -s "http://127.0.0.1:7777/metrics"  |grep test_c1 |grep jetstream_consumer_num_pending
jetstream_consumer_num_pending{account="USERS",cluster="nats_c",consumer_desc="",consumer_leader="nats-55",consumer_name="test_c1",domain="",is_consumer_leader="false",is_meta_leader="false",is_stream_leader="false",meta_leader="nats-55",server_id="http://x.x.x.64:8222",server_name="nats-64",stream_leader="nats-65",stream_name="test"} 0

# curl -s "http://127.0.0.1:7777/metrics"  |grep test_c1 |grep jetstream_consumer_num_pending
jetstream_consumer_num_pending{account="USERS",cluster="nats_c",consumer_desc="",consumer_leader="nats-55",consumer_name="test_c1",domain="",is_consumer_leader="false",is_meta_leader="false",is_stream_leader="true",meta_leader="nats-55",server_id="http://x.x.x.65:8222",server_name="nats-65",stream_leader="nats-65",stream_name="test"} 0

```

### Given the capability you are leveraging, describe your expectation?

On all nodes, the "num_pending" or "jetstream_consumer_num_pending" should be the same.

### Given the expectation, what is the defect you are observing?

On the grafana board, the pending messages of each nats-server node is inconsistent.

**Comment by derekcollison** [comment]
Only the leader peer will have the correct value. We suggest using the NATS cli for things like this vs the monitoring endpoint..

**Comment by yuzhou-nj** [comment]
ok.  curl -s "http://127.0.0.1:7777/metrics"  |grep test_c1 |grep jetstream_consumer_num_pending

**Comment by yuzhou-nj** [comment]
> Only the leader peer will have the correct value. We suggest using the NATS cli for things like this vs the monitoring endpoint..

Can the information be synchronized from the leader during the query?
All the cluster node information is displayed on the grafana board. This data inconsistency causes confusion.
Each consumer has its own leader. I do not know which node to query.

Thank you!

**Comment by ripienaar** [comment]
Prom data has a label that indicates the leader so you can easily graph just the leader

for monitoring it’s better to use the system account and natsCLI to make these requests rather than http done there you can ask the leader to respond

**Comment by wallyqs** [comment]
There have been many fixes to have these metrics aligned with the followers, most recent https://github.com/nats-io/nats-server/pull/5533/files  that is included in v2.10.17.  Closing for now.
